azure/durable_functions/models/utils/entity_utils.py

```diff
@@ -0,0 +1,91 @@
+class EntityId:
+    """EntityId.
+
+    It identifies an entity by its name and its key.
+    """
+
+    def __init__(self, name: str, key: str):
+        """Instantiate an EntityId object.
+
+        Identifies an entity by its name and its key.
+
+        Parameters
+        ----------
+        name: str
+            The entity name
+        key: str
+            The entity key
+
+        Raises
+        ------
+            ValueError: If the entity name or key are the empty string
+        """
+        if name == "":
+            raise ValueError("Entity name cannot be empty")
+        if key == "":
+            raise ValueError("Entity key cannot be empty")
+        self.name: str = name
+        self.key: str = key
+
+    @staticmethod
+    def get_scheduler_id(entity_id: 'EntityId') -> str:
+        """Produce a SchedulerId from an EntityId.
+
+        Parameters
+        ----------
+        entity_id: EntityId
+            An EntityId object
+
+        Returns
+        -------
+        str:
+            A SchedulerId representation of the input EntityId
+        """
+        return f"@{entity_id.name.lower()}@{entity_id.key}"
```

Comment:
We should make sure this lines up with the C# algorithm for this, as I remember there being some issues with which properties we lowercased or not and we changed it at some point. This is probably using the latest/correct version, but I would double check.

Reply:
Hmm, I'm not aware of this algorithmic change. I'll need some help verifying this is right. At the very least though, basic uses of entities do work with this implementation

Reply:
Looking at EntityId.cs, this looks correct. The change may have been a long time ago, or I could be thinking of some other identifier we were lower casing. Just wanted to double check.